src/microbithal_js.c

```diff
@@ -400,33 +403,33 @@ void microbit_hal_log_set_timestamp(int period) {
 }
 
 int microbit_hal_log_begin_row(void) {
+    return mp_js_hal_log_begin_row();
     /*
     if (uBit.log.beginRow() != DEVICE_OK) {
         return -1;
     }
     return 0;
     */
-    return 0;
 }
 
 int microbit_hal_log_end_row(void) {
+    return mp_js_hal_log_end_row();
     /*
     if (uBit.log.endRow() != DEVICE_OK) {
         return -1;
     }
     return 0;
     */
-    return 0;
 }
 
 int microbit_hal_log_data(const char *key, const char *value) {
+    return mp_js_hal_log_data(key, value);
     /*
     if (uBit.log.logData(key, value) != DEVICE_OK) {
         return -1;
     }
     return 0;
     */
-    return 0;
 }
```

Comment:
To get the right "log full" behaviour I think we'll need to be propagating errors here.